js/libs/core.js

```diff
@@ -617,39 +625,72 @@ Fliplet.FormBuilder = (function() {
       var hasOptions = component.props.options && Array.isArray(component.props.options.type());
       var hasSelectAll = component.props.addSelectAll && typeof component.props.addSelectAll.default === 'boolean';
       var isSlider = component.props._componentName.default === 'flSlider';
+      var isMatrix = component.props._componentName.default === 'flMatrix';
 
       // If options is an array, automatically deal with options
-      if (hasOptions) {
-        component.computed._options = function generateOptions() {
-          return this.options.map(function(option) {
-            if (option.id && option.label && option.id != option.label) {
-              return option.label + ' <' + option.id + '>';
-            }
+      if (hasOptions || isMatrix) {
+        if (isMatrix) {
```

Comment:
why do we again need check here?

Reply:
@yogeshmsharma-xoriant here same method will be called for Radio button and Checkbox options also that is only for matrix field validation has been added.

Reply:
The following part of the code keeps repeating. Can we make this DRY please?

```js
            .map(function(option) {
              if (option.id && option.label && option.id != option.label) {
                return option.label + ' <' + option.id + '>';
              }

              return option.label || option.id;
            }).join('\r\n')
```

I suggest a function as specified below:

```
/**
 * Generate text configurations for radio/checkbox options, separated by new lines
 * @param {Array} options - A list of options to be mapped
 * @returns {String} Text options for the configuration interface
 */
function generateOptionsAsText (options) {
}
```